File: Python/Programmers/Level 1/budget.py
def solution(d, budget):
    answer = 0

    d = sorted(d)

    for i in d:
        budget -= i
        if budget < 0:
            break
        answer += 1

    return answer


d = [2, 2, 3, 3]
budget = 10
print(solution(d, budget))

File: Java/src/Programmers/ProgrammersCodingTest/Budget.java
package Programmers.ProgrammersCodingTest;

import java.util.Arrays;

public class Budget {
    public int solution(int[] d, int budget) {
        int answer = 0;

        Arrays.sort(d);

        for(int i = 0; i < d.length; i++) {
            if(budget - d[i] >= 0) {
                budget -= d[i];
                answer++;
            } else {
                i++;
            }
        }
        return answer;
    }

    public static void main(String[] args) {
        Budget solution = new Budget();

        int[] d = {1, 3, 2, 5, 4};
        int budget = 9;

        System.out.println(solution.solution(d, budget));
    }
}